src/server/resources/Observation.js

```diff
@@ -78,6 +78,16 @@ class Observation extends DomainResource {
 		return this._subject;
 	}
 
+	// Retrieve the patient id from the reference
+	get patientId() {
+		// subject.reference = 'Patient/{patientId}'
+		return (
+			this._subject
+			&& this._subject.reference
+			&& this._subject.reference.split('/')[1]
```

Comment:
Would it make sense to do additional validity checks; i.e. make sure that the reference is actually in this format?

Reply:
If the format is not this then the data is not conforming to the spec. I don't think we should add additional logic to parse invalid formats.